neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/remote/http/transaction/impl/HttpNeo4jResourceLocalTransactionCoordinator.java

```diff
@@ -330,7 +330,7 @@ public void markRollbackOnly() {
 
 		@Override
 		public Object getTransactionId() {
-			return tx.getId();
+			return tx != null ? tx.getId() : null;
```

Comment:
Transaction is mandatory.

What kind of error we show in case a transaction is not present? If it's not clear enough we can improve it. Anyway, this would be a separate issue